src/components/image-plus.js

```diff
@@ -227,22 +192,49 @@ AFRAME.registerComponent("image-plus", {
           texture = await this.loadGIF(url);
         } else if (contentType.startsWith("image/")) {
           texture = await this.loadImage(url);
-        } else if (contentType.startsWith("video")) {
+        } else if (contentType.startsWith("video/") || contentType.startsWith("audio/")) {
           texture = await this.loadVideo(url);
         } else {
-          throw new Error(`Unknown centent type: ${contentType}`);
+          throw new Error(`Unknown content type: ${contentType}`);
         }
 
         textureCache.set(url, { count: 1, texture });
       }
+
+      if (texture.audioSource) {
+        const sound = new THREE.PositionalAudio(this.el.sceneEl.audioListener);
+        sound.setNodeSource(texture.audioSource);
+        this.el.setObject3D("sound", sound);
+      }
     } catch (e) {
       console.error("Error loading media", this.data.src, e);
       texture = errorTexture;
     }
 
-    const material = this.el.getObject3D("mesh").material;
+    const material = new THREE.MeshBasicMaterial();
+    material.side = THREE.DoubleSide;
+    material.transparent = true;
     material.map = texture;
     material.needsUpdate = true;
-    this._fit(texture.image.videoWidth || texture.image.width, texture.image.videoHeight || texture.image.height);
+    material.map.needsUpdate = true;
+
+    const ratio =
+      (texture.image.videoHeight || texture.image.height || 1.0) /
+      (texture.image.videoWidth || texture.image.width || 1.0);
+    const width = Math.min(1.0, 1.0 / ratio);
+    const height = Math.min(1.0, ratio);
+
+    const geometry = new THREE.PlaneGeometry(width, height, 1, 1);
+    this.mesh = new THREE.Mesh(geometry, material);
+    this.el.setObject3D("mesh", this.mesh);
+    this.el.setAttribute("shape", {
+      shape: "box",
+      halfExtents: { x: width / 2, y: height / 2, z: 0.05 }
+    });
+    if (this.el.components.body && this.el.components.body.body) {
+      this.el.components.body.syncToPhysics(); // not sure if necessary?
```

Comment:
This doesn't seem like it should be necessary, but verify?

Reply:
Since we have already tested it a bunch with this in, going to leave it in for now to avoid missing some edge case. Put in a todo to evaluate though.